### DISCUSSION TITLE
群友提问，关于nerf实现中recenter_poses的疑问

### DISCUSSION THREAD

**Discussion by yangjiheng** [yangjiheng/nerf_and_beyond_docs#2] [General]
From @qc:

想请教各位大佬一个比较基础的问题
原始的nerf里，在load_llff.py里使用了recenter_poses函数，我一直没有理解它的作用，尤其是下面这行代码
poses = np.linalg.inv(c2w) @ poses 
poses本身就是一个c2w矩阵，再乘上一个w2c矩阵，结果应该是一个c2c的矩阵，但是代码中将这个结果一直作为c2w在使用。这个是否合理？
而且，我可视化了这个结果，原本llff数据下，相机朝向和世界坐标系的z轴是相反的，而且相机分散在整个空间内，围绕这场景，经过recenter_poses后，所有相机都挪到了世界坐标系原点，朝向和世界坐标系的z轴相同了，这合理吗？

------Context------
Jason: 总记得有过一个讨论，看看这个是不是有帮助：https://github.com/bmild/nerf/issues/34
qc: 看过这个帖子，有个跟帖还是我发的，但是还是没太理解

**Comment by hjxwhy** [comment]
只是把世界坐标系换一下而已，没什么不合理的，原来的相机相对位置还是不变的，坐标系本来就是相对的

**Comment by yangjiheng** [comment]
From Hugo,

个人理解是 这里的poses = np.linalg.inv(c2w) @ poses 最左边的poses是在ndc coordinate也就是 [-1, 1]这个坐标系下相机的坐标, 最右边的poses是gt 尺度下面的相机坐标系的坐标, 中间的c2w 是gt尺度下相机坐标的均值, 也就是ndc坐标系下的原点, 类比相机坐标系到世界坐标系的转换, np.linalg.inv(c2w) 可以简单理解成 从gt尺度坐标系到ndc坐标系的转换。转换到ndc coordinate的原因 我之前跟人讨论的时候好像听过一个结论是 xyz放缩到[-1,1]区间内positional encoding比较好学 具体原因有点忘了。

**Comment by yangjiheng** [comment]
From Mazart,

这里c2w算的是poses的平均位姿，也就是一个变换矩阵，假设这个平均位姿c2w的逆左乘c2w后，得到的矩阵R为单位阵，T为0，也就是原点位置。poses = np.linalg.inv(c2w) @ poses所以这个代码执行后，poses的平均位姿就处在世界坐标系原点。也就完成了归一化。1 2 3 4 5平均数为3，都减去平均数3后，平均数就变为0了，就归一化了
